api/capi/src/nnstreamer-capi-pipeline.c

```diff
@@ -1671,103 +1671,271 @@ ml_pipeline_element_get_handle (ml_pipeline_h pipe, const char *element_name,
  * @brief Releases the given element handle.
  */
 int
-ml_pipeline_element_release_handle (ml_pipeline_element_h elm_h)
+ml_pipeline_element_release_handle (ml_pipeline_element_h elem_h)
 {
-  handle_init (common_elem, elm_h);
+  handle_init (common_elem, elem_h);
 
   elem->handles = g_list_remove (elem->handles, common_elem);
   g_free (common_elem);
 
-  handle_exit (elm_h);
+  handle_exit (elem_h);
 }
 
 /**
- * @brief Sets element properties in NNStreamer pipelines.
+ * @brief Check property existence and its type.
  */
-int ml_pipeline_element_set_property (ml_pipeline_element_h elm_h,
-    const char *first_property_name, ...)
+static bool
+ml_pipeline_element_check_property (GObjectClass *class, const char *property_name, const GType type)
 {
-  va_list args_check;
-  va_list args;
-  const char *name;
+  GParamSpec *pspec = NULL;
 
-  handle_init (common_elem, elm_h);
+  /* Check property existence */
+  pspec = g_object_class_find_property (class, property_name);
+  if (pspec == NULL) {
+    ml_loge ("The property name [%s] does not exist.", property_name);
+    return FALSE;
+  }
+
+  /* Compare property's type with given type */
+  if (!((pspec->value_type == type) ||
+    (type == G_TYPE_ENUM && G_TYPE_IS_ENUM (pspec->value_type)) ||
+    (type == G_TYPE_INT64 && pspec->value_type == G_TYPE_LONG) ||
+    (type == G_TYPE_UINT64 && pspec->value_type == G_TYPE_ULONG) ||
+    (type == G_TYPE_DOUBLE && pspec->value_type == G_TYPE_FLOAT))) {
```

Comment:
value-type is G_TYPE_DOUBLE or G_TYPE_FLOAT?

Reply:
It's OK since I alreay check `pspec->value_type == type` first.